theseus/theseus_layer.py

```diff
@@ -93,3 +118,154 @@ def device(self) -> torch.device:
     @property
     def dtype(self) -> torch.dtype:
         return self.objective.dtype
+
+
+def _forward(objective, optimizer, optimizer_kwargs, input_data):
```

Comment:
Any reason this function should be outside of the `TheseusLayer` class? We could move it inside and rename to `_forward_impl()` to follow what we do in many other places.